waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -623,38 +623,41 @@ method init*(g: OnchainGroupManager): Future[void] {.async.} =
-                                                     query = keystoreQuery,
-                                                     appInfo = RLNAppInfo)
-    if keystoreCredRes.isErr():
-      raise newException(CatchableError, "could not parse the keystore: " & $keystoreCredRes.error)
-    let keystoreCred = keystoreCredRes.get()
-    g.membershipIndex = some(keystoreCred.treeIndex)
-    when defined(rln_v2):
-      g.userMessageLimit = some(keystoreCred.userMessageLimit)
-    # now we check on the contract if the commitment actually has a membership
-    try:
-      let membershipExists = await rlnContract.memberExists(keystoreCred
-                                                            .identityCredential
-                                                            .idCommitment.toUInt256()).call()
-      if membershipExists == 0:
-        raise newException(CatchableError, "the provided commitment does not have a membership")
-    except CatchableError:
-      raise newException(CatchableError, "could not check if the commitment exists on the contract: " &
-                                         getCurrentExceptionMsg())
-
-    g.idCredentials = some(keystoreCred.identityCredential)
+      if g.membershipIndex.isSome():
+        keystoreQuery.treeIndex = MembershipIndex(g.membershipIndex.get())
+      waku_rln_membership_credentials_import_duration_seconds.nanosecondTime:
+        let keystoreCredRes = getMembershipCredentials(path = g.keystorePath.get(),
+                                                      password = g.keystorePassword.get(),
+                                                      query = keystoreQuery,
+                                                      appInfo = RLNAppInfo)
+      if keystoreCredRes.isErr():
+        raise newException(CatchableError, "could not parse the keystore: " & $keystoreCredRes.error)
+      let keystoreCred = keystoreCredRes.get()
+      g.membershipIndex = some(keystoreCred.treeIndex)
+      when defined(rln_v2):
+        g.userMessageLimit = some(keystoreCred.userMessageLimit)
+      # now we check on the contract if the commitment actually has a membership
+      try:
+        let membershipExists = await rlnContract.memberExists(keystoreCred
+                                                              .identityCredential
+                                                              .idCommitment.toUInt256()).call()
+        if membershipExists == 0:
+          raise newException(CatchableError, "the provided commitment does not have a membership")
+      except CatchableError:
+        raise newException(CatchableError, "could not check if the commitment exists on the contract: " &
+                                          getCurrentExceptionMsg())
+
+      g.idCredentials = some(keystoreCred.identityCredential)
+    else:
+      warn "File provided as keystore path does not exist", path=g.keystorePath.get() 
```

Comment:
probably want to exit here right, this means the node will not be able to generate proofs.
additionally, can we add a test vector to further ensure that it works as intended?

Reply:
Perfect - thought we could just continue relaying messages as it's supposed to be optional. But if path and password are passed, I guess that the intention is to be able to generate proofs.

This implies that in nwaku-compose generating the credentials will be mandatory. Or in other words, we won't be able to run a node by simply running `docker-compose up -d` without generating credentials and without manually changing `run_node.sh`